cfpb/cfgov-refresh
# Bring Google Tag Manager custom HTML tags into cfgov-refresh

Google Tag Manager (GTM) introduces JS code to the site without code review and tests. 
GTM also includes youtube code that may be the source of [ghe]/CFGOV/platform/issues/166.

This PR brings the existing code from GTM to this repo so that we can improve its quality and test coverage. This will run the scripts through webpack and make them available on the site at the static/apps/analytics-gtm/ URL, which will allow GTM to lazy load the scripts by utilizing the global `jsl()` function we already use to load other scripts.

## Additions

- GTM custom HTML tags from Google Tag Manager dashboard.

## Changes

- Moves `javascript_loader` jinja block to the head above the analytics code imports. 

## Testing

1. JS should load on the site as before. 
2. Edit `unprocessed/apps/analytics-gtm/bah-check-rates-listeners.js` to add `console.log( 'OAHRCAnalytics loaded!' );` on line 2. 
3. Run `gulp build`.
4. Add this:
 ```js 
 <script>
    jsl(['/static/apps/analytics-gtm/js/bah-check-rates-listeners.js']);
 </script>
```
To line 150 of `base-common.html`
5. Verify you see `OAHRCAnalytics loaded!` in the dev console when loading a page. **Note: there will be an error too, unless you're viewing on the appropriate OAH page. This doesn't matter though, we just want to know that it loads.**

## Notes

- After this goes to production the custom HTML tags in GTM should be updated with `jsl(['/static/apps/analytics-gtm/js/[script to load].js']);`, where `[script to load]` is the custom HTML tag to load.
